common/datastore/conversions.go

```diff
@@ -44,7 +44,7 @@ func toObject(raw_data interface{}, definition DataStoreDefinition) (interface{}
 	unfiltered_data, ok := raw_data.(map[string]interface{})
 
 	if !ok {
-		return nil, ErrTypeMismatch
+		return nil, NewErrTypeMismatch(raw_data, "[]interface{}")
```

Comment:
Should this be "map[string]interface{}" instead of "[]interface{}"?

Reply:
Good catch, thanks